### ISSUE TITLE
Understanding duckdb WKB geometry type (not readable by shapely)

### ISSUE THREAD

**Issue by ncclementi** [duckdb/duckdb_spatial#188]
On Ibis we are working on supporting the duckdb geospatial spatial extension (see https://github.com/ibis-project/ibis/pull/7454) and we would like to understand better what's coming out of geometry types when we select from a column. 

The docs say it's WKB but when we try to convert to a shapely object, we get an error. But when we call `st_aswkb` then we can get it to work, but it doesn’t seem like this should be necessary. 

Here is a reproducer that @cpcloud put together that shows this: 
https://gist.github.com/cpcloud/82d98fb1bd1a4919c6a4c41643ad3141

**Comment by Maxxen** [comment]
Hi!

DuckDB's internal representation of the geometry type is not actually WKB, in fact it is very similar to PostGIS geometry representation, although with a different "header". However **I can't promise that our geometry representation is fully stable yet**. I still want to extend it (soon!) so that we can store for example multi-dimensional geometries (Z/M), SRID and/or other properties like whether a geometry is valid/solid/geodetic. In theory we should be able to do this in a backwards compatible way but again, no promises.

Here's some excerpts from the code that hopefully provides some insight into how the format works.

The header: (4 bytes)
https://github.com/duckdb/duckdb_spatial/blob/b4cb5bd214c53f65a0d9f585fb9490bd50cbb4bf/spatial/include/spatial/core/geometry/geometry.hpp#L431-L435

The "properties" (1 byte)
https://github.com/duckdb/duckdb_spatial/blob/b4cb5bd214c53f65a0d9f585fb9490bd50cbb4bf/spatial/include/spatial/core/geometry/geometry_properties.hpp#L8-L16

The full geometry. (always a multiple of 8 bytes)
The comments here are slightly outdated, if the geometry is non-empty and not a point there is a 8xF32 bounding box serialized after the "header" and the 4 byte padding.

https://github.com/duckdb/duckdb_spatial/blob/b4cb5bd214c53f65a0d9f585fb9490bd50cbb4bf/spatial/src/spatial/core/geometry/geometry_factory.cpp#L134-L161

Full Deserialization code

https://github.com/duckdb/duckdb_spatial/blob/b4cb5bd214c53f65a0d9f585fb9490bd50cbb4bf/spatial/src/spatial/core/geometry/geometry_factory.cpp#L523-L691

**Comment by janneskruse** [comment]
@Maxxen Yes `duckdb                    1.3.1                    pypi_0    pypi`

Same for this btw:
```
con.execute(f"""
    SELECT
        id,
        names.primary as name,
        confidence,
        CAST(socials AS JSON) as socials,
        categories.primary as category,
        st_asgeojson(ST_GeomFromWKB(ST_AsWKB(geometry))) as geometry
    FROM read_parquet('s3://overturemaps-us-west-2/release/2025-05-21.0/theme=places/type=place/*', filename=true, hive_partitioning=1)
    LIMIT {limit}
""").fetchall()
```

Which is even faster I noticed but still produces incorrect coordinates:
```
[('08ff39b3492ea01103146f4f33a16d8d',
  'New National Pipe Store',
  0.8490566037735849,
  '["https://www.facebook.com/105459768169555"]',
  'event_planning',
  '{"type":"Point","coordinates":[7.02349128656905e219,2.4978251817949645e136]}'),
 ('08ff3a1653d759730329e616a052a0e3',
  'แหลมใหญ่แล็บ คลินิกเทคนิคการแพทย์ สมุทรสงคราม',
  0.7866439086835929,
  '["https://www.facebook.com/111694701281593"]',
  'health_and_medical',
  '{"type":"Point","coordinates":[2.7932399778917712e113,3.572552393542608e-79]}'),
 ('08ff3aa729c6eb20030e9ba5c1185a07',
  'Chittagong Victory National School-CVNS',
  0.8712799768852932,
  '["https://www.facebook.com/100714742574676"]',
  'school',
  '{"type":"Point","coordinates":[9.89256809286431e42,3.936063553343023e-125]}')
```

**Comment by janneskruse** [comment]
@Maxxen Thanks for directly jumping in. I just reinstalled v1.3.1 without cache but the issue remains. Should I install a specific version?

**Comment by Maxxen** [comment]
The original issue here has been resolved, and DuckDB v1.5 now uses WKB internally so the original answer isn't really relevant anymore.

We don't support native geoarrow yet.
Feel free to open a new issue if you have another problem!
